std/internal/entropy.d

```diff
@@ -0,0 +1,889 @@
+            ++timesFailed;
+
+    assert(
+        timesFailed <= 1,
+        "Suspicious random data: Potential security issue or really unlucky; please retry."
+    );
+}
+
+@nogc nothrow:
+
+// Flagship function
+/++
+    Retrieves random data from an applicable system CSPRNG.
+
+    Params:
+        buffer = An output buffer to store the retrieved entropy in.
+                 The length of it will determine the amount of random data to
+                 be obtained.
+
+                 This function (and all overloads) always attempt to fill
+                 the entire buffer. Therefore, they can block, spin or report
+                 an error.
+
+    Returns:
+        An `EntropyResult` that either reports success
+        or the type of error that has occurred.
+
+        In case of an error, the data in `buffer` MUST NOT be used.
+        The recommended way to check for success is through the `isOK()`
+        helper function.
+ +/
+EntropyResult getEntropy(scope void[] buffer) @safe
+{
+    return getEntropyImpl(buffer);
+}
+
+///
+@safe unittest
+{
+    int[4] bytes;
+    if (getEntropy(cast(void[]) bytes).isOK)
+    {
+        // Success; data in `bytes` may be used.
+    }
+
+    assert((cast(void[]) bytes).length == bytes.sizeof);
+}
+
+// Convenience overload
+/// ditto
+EntropyResult getEntropy(scope ubyte[] buffer) @safe
+{
+    return getEntropy(cast(void[]) buffer);
+}
+
+///
+@safe unittest
+{
+    ubyte[16] bytes;
+    if (getEntropy(bytes).isOK)
+    {
+        // Success; data in `bytes` may be used.
+    }
+}
+
+// Convenience wrapper
+/// ditto
+/++
+    Retrieves random data from an applicable system CSPRNG.
+
+    Params:
+        buffer = An output buffer to store the retrieved entropy in.
+        length = Length of the provided `buffer`.
+                 Specifying a wrong value here, will lead to memory corruption.
+ +/
+EntropyResult getEntropy(scope void* buffer, size_t length) @system
+{
+    return getEntropy(buffer[0 .. length]);
+}
+
+///
+@system unittest
+{
+    ubyte[16] bytes;
+    if (getEntropy(cast(void*) bytes.ptr, bytes.length).isOK)
+    {
+        // Success; data in `bytes` may be used.
+    }
+}
+
+///
+@system unittest
+{
+    int number = void;
+    if (getEntropy(&number, number.sizeof).isOK)
+    {
+        // Success; value of `number` may be used.
+    }
+}
+
+/++
+    Manually set the entropy source to use for the current thread.
+
+    As a rule of thumb, this SHOULD NOT be done.
+
+    It might be useful in cases where the default entropy source — as chosen by
+    the maintainer of the used compiler package — is unavailable on a system.
+    Usually, `EntropySource.tryAll` will be the most reasonable option
+    in such cases.
+
+    Examples:
+
+    ---
+    // Using `forceEntropySource` almost always is a bad idea.
+    // As a rule of thumb, this SHOULD NOT be done.
+    forceEntropySource(EntropySource.none);
+    ---
+ +/
+void forceEntropySource(EntropySource source) @safe
```

Comment:
```
std/internal/entropy.d(228:6)[warn]: Public declaration 'forceEntropySource' has no documented example.
```
No, that’s wrong you incompetent piece of sh…software!

Reply:
I’ve decided to disable this style check: <https://github.com/dlang/phobos/pull/10748/commits/cfcb0ced23a6f3168e836c647a72e6f2069e75f9>